src/pages/PostCreate.jsx

```diff
@@ -1,4 +1,173 @@
+import { useState } from "react";
+import { useNavigate } from "react-router-dom";
+import { usePosts } from "../context/PostsContext";
+import categories from "../util/categories";
+
 export default function PostCreate() {
-    return <h1>Create Post Page</h1>;
-  }
-  
\ No newline at end of file
+  const navigate = useNavigate();
+  const { createPost } = usePosts();
+
+  const [formData, setFormData] = useState({
+    title: "",
+    description: "",
+    category: "",
+    location: "",
+    photo: "",
+    canDeliver: false,
+  });
+
+  // Handle text input change
+  const handleChange = (e) => {
+    const { name, value, type, checked } = e.target;
+    setFormData((prev) => ({
+      ...prev,
+      [name]: type === "checkbox" ? checked : value,
+    }));
+  };
+
+  // Handle image upload for local preview
+  const handleImageChange = (e) => {
```

Comment:
since the photo change is not handled by the `handleChange` function, I'd separate the states, use one state for text and boolean inputs and have another useState for photos, especially since we're targetting to allow the user to upload many photos

something like
```javascript
const [photos, setPhotos] = useState([])

when adding a photo
... newPhoto = { url, URL.create..., photo: photo}
setPhotos( [...photos, newPhoto]

also I'd add a handler to remove a photo that the user doesn't want to upload
```